analysis/filter/StructureModifier/src/main/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/FolderMover.kt

```diff
@@ -58,7 +57,17 @@ class FolderMover(private val project: Project, private val projectName: String?
 
     private fun insertInNewStructure(destinationPath: List<String>, node: MutableNode) {
         if (destinationPath.isEmpty()) {
-            node.children.addAll(toMove!!)
+            val destinationNodeNamesAndType: HashMap<String, NodeType?> = hashMapOf()
+            node.children.forEach { destinationNodeNamesAndType[it.name] = it.type }
+
+            val filteredNodesToMove = toMove!!.filter {
+                !(destinationNodeNamesAndType.containsKey(it.name) && destinationNodeNamesAndType[it.name] == it.type)
+            }
+            if (filteredNodesToMove.size < toMove!!.size) {
+                logger.warn("Some nodes are already available in the target location, they were not moved and discarded instead.")
+            }
+            node.children.addAll(filteredNodesToMove)
```

Comment:
I think this is a better solution 👍 but, what if the children with the same name+type have a different content? Should there be an option to rename the new child, instead of deleting it?

Reply:
Well here is what I thought when implementing this:

This filter aims mostly at correcting non-matching hierarchies, so that the .cc.json files can be merged, therefore the major use case will be something like moving nodes from /root/ to /root/foobar/ or from /root/foo/ to /root/bar (in case the folder has two different names) or vice versa. In most cases I suppose the target folder will not exist in the original structure.

Renaming the new child (foo -> foo_copy or similar) does, in my opinion not make sense in most use cases, since then we do not get matching hierarchies either, because foo_copy cannot be merged with foo.

If there is really the case where a user wants to move /something to /else and else and have matching children, merging them would probably be the most useful way. However I am reluctant to implement something like this, because I don't want to duplicate the merge-filter here, also i think the use case is very rare. If the use case occurs however, the two filters combined allow to solve the issue.

Therefore I think that discarding & warning is probably the best way to implement this here

Reply:
Alright, yea that makes sense 👍